### PR TITLE
Setup CI/CD Pipeline

### PR DESCRIPTION
**Pull request by raphaelseher**
Project Setup Steps updated: https://github.com/SE2-Gwent/Gwent/wiki#project-setup

Documentation: https://github.com/SE2-Gwent/Gwent/wiki/CI-CD

### CODE REVIEW

**Comment by raphaelseher** [comment]
> There is no coverage information present for the Files changed

|Total Project Coverage|0%|:x:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
> There is no coverage information present for the Files changed

|Total Project Coverage|0%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
> There is no coverage information present for the Files changed

|Total Project Coverage|0%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
> There is no coverage information present for the Files changed

|Total Project Coverage|0%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
> There is no coverage information present for the Files changed

|Total Project Coverage|0%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
> There is no coverage information present for the Files changed

|Total Project Coverage|0%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
> There is no coverage information present for the Files changed

|Total Project Coverage|0%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
> There is no coverage information present for the Files changed

|Total Project Coverage|0%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
> There is no coverage information present for the Files changed

|Total Project Coverage|0%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
> There is no coverage information present for the Files changed

|Total Project Coverage|0%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
> There is no coverage information present for the Files changed

|Total Project Coverage|0%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
> There is no coverage information present for the Files changed

|Total Project Coverage|0%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
> There is no coverage information present for the Files changed

|Total Project Coverage|0%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
> There is no coverage information present for the Files changed

|Total Project Coverage|0%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
> There is no coverage information present for the Files changed

|Total Project Coverage|0%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
> There is no coverage information present for the Files changed

|Total Project Coverage|0%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
> There is no coverage information present for the Files changed

|Total Project Coverage|0%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
> There is no coverage information present for the Files changed

|Total Project Coverage|0%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
> There is no coverage information present for the Files changed

|Total Project Coverage|0%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
> There is no coverage information present for the Files changed

|Total Project Coverage|0%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
> There is no coverage information present for the Files changed

|Total Project Coverage|0%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
|File|Coverage [0%]|:green_apple:|
|:-|:-:|:-:|
|[SettingsFragment.java](https://github.com/SE2-Gwent/Gwent/blob/ca6f08fe182b5a83c7030ca49fceb674ad070aee/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fsettings%2FSettingsFragment.java)|0%|:green_apple:|
|[SettingsViewModel.java](https://github.com/SE2-Gwent/Gwent/blob/ca6f08fe182b5a83c7030ca49fceb674ad070aee/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fsettings%2FSettingsViewModel.java)|0%|:green_apple:|
|[MainActivity.java](https://github.com/SE2-Gwent/Gwent/blob/ca6f08fe182b5a83c7030ca49fceb674ad070aee/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2FMainActivity.java)|0%|:green_apple:|

|Total Project Coverage|0%|:green_apple:|
|:-|:-:|:-:|

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+# separate terms of service, privacy policy, and support
+# documentation.
+# This workflow will build a Java project with Gradle and cache/restore any dependencies to improve the workflow execution time
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: Generate signed APK
+
+on:
+  pull_request:
+    branches: [develop]
```
Comment (Sikredo) [inline .github/workflows/deploy.yml:12]: Should we change the branch for the APK Generation to main?

**Comment by raphaelseher** [comment]
|File|Coverage [0%]|:green_apple:|
|:-|:-:|:-:|
|[SettingsFragment.java](https://github.com/SE2-Gwent/Gwent/blob/48f0af03a931a7547d7202f34c0f20f3c7874f07/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fsettings%2FSettingsFragment.java)|0%|:green_apple:|
|[SettingsViewModel.java](https://github.com/SE2-Gwent/Gwent/blob/48f0af03a931a7547d7202f34c0f20f3c7874f07/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fsettings%2FSettingsViewModel.java)|0%|:green_apple:|
|[MainActivity.java](https://github.com/SE2-Gwent/Gwent/blob/48f0af03a931a7547d7202f34c0f20f3c7874f07/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2FMainActivity.java)|0%|:green_apple:|

|Total Project Coverage|0%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
I rebased my commits to cleanup the history because of all the commits I did to trigger the actions.

**Pull request merged**